src/Uno.UI/UI/Xaml/Application.cs

```diff
@@ -185,35 +185,9 @@ internal void SetRequestedTheme(ApplicationTheme requestedTheme)
 
 		private void OnRequestedThemeChanged()
 		{
-			if (Windows.UI.Xaml.Window.Current.Content is FrameworkElement root)
-			{
-				PropagateThemeChanged(root);
-			}
-
-			void PropagateThemeChanged(object instance)
-			{
-				// Update ThemeResource references that have changed
-				if (instance is FrameworkElement fe)
-				{
-					fe.UpdateThemeBindings();
-				}
-
-				//Try Panel.Children before ViewGroup.GetChildren - this results in fewer allocations
-				if (instance is Controls.Panel p)
-				{
-					foreach (object o in p.Children)
-					{
-						PropagateThemeChanged(o);
-					}
-				}
-				else if (instance is ViewGroup g)
-				{
-					foreach (object o in g.GetChildren())
-					{
-						PropagateThemeChanged(o);
-					}
-				}
-			}
+			FrameworkElement.PropagateThemeScope(
+				Windows.UI.Xaml.Window.Current.Content,
```

Comment:
pretty edge-case scenario, but what happens if by some chance RequestedTheme has been set on `Windows.UI.Xaml.Window.Current.Content`?

Reply:
That is actually covered - the second argument for the `PropagateThemeScope` method just says what is the requested theme of the outside scope, if any. Inside the `UpdateActualTheme` method is called, which checks the processed element's `RequestedTheme` and gives it precedence.